baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ForbidJavaxParameterType.java

```diff
@@ -0,0 +1,133 @@
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.TypeSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import java.lang.annotation.Annotation;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "Supplying an object which uses legacy javax types, such as javax.ws.rs to a\n"
+                + "method which requires newer jakarta types is a runtime error. This check ensures\n"
+                + "that you only supply proper types to these methods which generally just take an\n"
+                + "untyped Object. There is no auto-fix for this check, you must fix it manually")
+public final class ForbidJavaxParameterType extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Matcher<ClassTree> HAS_JAXRS_ANNOTATIONS = Matchers.hasMethod(Matchers.anyOf(
+            Matchers.hasAnnotation("javax.ws.rs.GET"),
+            Matchers.hasAnnotation("javax.ws.rs.POST"),
+            Matchers.hasAnnotation("javax.ws.rs.DELETE"),
+            Matchers.hasAnnotation("javax.ws.rs.PUT"),
+            Matchers.hasAnnotation("javax.ws.rs.HEAD"),
+            Matchers.hasAnnotation("javax.ws.rs.OPTIONS")));
+
+    private static final Matcher<ClassTree> IMPLEMENTS_FEATURE = Matchers.isSubtypeOf("javax.ws.rs.core.Feature");
+
+    private static final Matcher<ClassTree> HAS_PATH_ANNOTATION = Matchers.hasAnnotation("javax.ws.rs.Path");
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        List<? extends ExpressionTree> arguments = tree.getArguments();
+        MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
+        List<VarSymbol> typeParameters = methodSymbol.getParameters();
+
+        if (typeParameters == null
+                || typeParameters.isEmpty()
+                || arguments == null
+                || arguments.isEmpty()
+                || arguments.size() != typeParameters.size()) {
+            return Description.NO_MATCH;
+        }
+
+        for (int i = 0; i < typeParameters.size(); ++i) {
+            VarSymbol parameter = typeParameters.get(i);
+            ExpressionTree argument = arguments.get(i);
+
+            Annotation maybeRequireJakarta = parameter.getAnnotation(ForbidJavax.class);
+            if (maybeRequireJakarta == null) {
+                continue;
+            }
+
+            if (hasJavaxInclusions(ASTHelpers.getSymbol(argument), state)) {
```

Comment:
This might be a bit easier if we use `ASTHelpers.getResultType(argument)` which handles the method/ctor return type indirection

Reply:
👍